tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/CachedBasicPropertiesHelper.cs

```diff
@@ -1,23 +1,45 @@
-﻿// <copyright file="CachedBasicPropertiesHelper.cs" company="Datadog">
+// <copyright file="CachedBasicPropertiesHelper.cs" company="Datadog">
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
 #nullable enable
-using Datadog.Trace.Util;
+using System;
+using System.Reflection.Emit;
+using Datadog.Trace.DuckTyping;
 
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.RabbitMQ;
 
 internal static class CachedBasicPropertiesHelper<TBasicProperties>
 {
-    // ReSharper disable once StaticMemberInGenericType
-    private static readonly ActivatorHelper Activator;
+    private static readonly Func<object, TBasicProperties> Activator;
 
     static CachedBasicPropertiesHelper()
     {
-        Activator = new ActivatorHelper(typeof(TBasicProperties).Assembly.GetType("RabbitMQ.Client.BasicProperties")!);
+        var targetType = typeof(TBasicProperties).Assembly.GetType("RabbitMQ.Client.BasicProperties")!;
```

Comment:
nit: errors in the static ctor are tricky beasts as they tend to not be invoked and cause weird errors. I'd be inclined to wrap everything in a try-catch, make the `Activator` property nullable, return `default` from `CreateHeaders()` in that case, and handle the null case properly at the callsite

Reply:
> errors in the static ctor are tricky beasts

agreed, thanks!

Reply:
Done, I think? Not super happy with the results 😅

Reply:
Yeah, it's ugly, but prob about the best we can do 🤷‍♂️ 😅